pkg/plugins/runtime/k8s/util/util.go

```diff
@@ -122,8 +123,8 @@ func FindPort(pod *kube_core.Pod, svcPort *kube_core.ServicePort) (int, *kube_co
 	return 0, nil, fmt.Errorf("no suitable port for manifest: %s", pod.UID)
 }
 
-func FindContainerStatus(pod *kube_core.Pod, containerName string) *kube_core.ContainerStatus {
-	for _, cs := range pod.Status.ContainerStatuses {
+func FindContainerStatus(containerName string, status []kube_core.ContainerStatus, otherStatuses ...[]kube_core.ContainerStatus) *kube_core.ContainerStatus {
```

Comment:
nit: this func definition is kind of weird. Can we just pass one slice here and do append if needed when we call it?

Reply:
Suspected this might be controversial...
I agree the definition is odd at first glance but I kind of like how it looks when calling it
```
util_k8s.FindContainerStatus(
  util_k8s.KumaSidecarContainerName,
  pod.Status.ContainerStatuses, 
  pod.Status.InitContainerStatuses,
)

util_k8s.FindContainerStatus(
  util_k8s.KumaSidecarContainerName,
  pod.Status.ContainerStatuses, 
)
```

I think I'll just expose two functions `FindContainerStatus` and `FindContainerOrInitContainerStatus` instead

Reply:
that works as well. ok.